src/components/scenes/WcWalletConnectScene.js

```diff
@@ -0,0 +1,192 @@
+    return { tempWallet, currencyWallets, wcEnabledWalletIds }
+  })
+
+  const createWcConnectionInfo = (connectedWalletId: string, uri: string): WcConnectionInfo => {
+    return {
+      // TODO: Placeholder data - dAppName, dAppUrl, timeConnected should come from parsed data in accountbased
+      dAppName: 'dAppName',
+      dAppUrl: 'dAppUrl',
+      timeConnected: 'timeConnected',
+      walletName: currencyWallets[connectedWalletId].name ?? 'NA',
+      walletId: connectedWalletId,
+      uri: uri
+    }
+  }
+
+  // Populate the list of dAppConnections
+  const dAppConnections = []
+  if (wcEnabledWalletIds.length > 0) {
+    for (const enabledWalletId of wcEnabledWalletIds) {
+      const connections = currencyWallets[enabledWalletId].otherMethods.wcGetConnections()
+      if (connections.length > 0) {
+        console.log(connections)
+        for (const uri of connections) {
+          dAppConnections.push(createWcConnectionInfo(enabledWalletId, uri))
+        }
+      }
+    }
+  }
+
+  const onScanSuccess = async (qrResult: string) => {
+    if (tempWallet === undefined) showError(new Error('onScanSuccess called without a defined wallet'))
+    else {
+      try {
+        const parsedScan = await tempWallet.parseUri(qrResult, tempWallet.currencyInfo.currencyCode)
+        const uriStr = parsedScan.walletConnect?.uri
+        // TODO: Call some other method (connect?) to allow accountbased to populate peer metadata
+        if (uriStr === undefined) throw new Error('Undefined uri in parsed scan')
+        navigation.navigate('wcConfirmation', { dAppName: 'dAppName', wcQRUri: uriStr })
+      } catch (error) {
+        showError(error)
+      }
+    }
+  }
+
+  const handleActiveConnectionPress = (wcConnectionInfo: WcConnectionInfo) => {
+    navigation.navigate('wcDetails', { wcConnectionInfo })
+  }
+
+  const handleNewConnectionPress = () => {
+    Airship.show(bridge => <ScanModal bridge={bridge} title={s.strings.scan_qr_label} />)
+      .then((result: string | void) => {
+        console.log(result)
+        if (result) {
+          onScanSuccess(result)
+        } else {
+          showError('No scan result')
+        }
+      })
+      .catch(error => {
+        showError(error)
+      })
+  }
+
+  return (
+    <SceneWrapper background="theme" hasTabs={false}>
+      <SceneHeader underline title={s.strings.wc_walletconnect_title} />
+      <ScrollView style={styles.container}>
+        <EdgeText style={styles.subTitle}>{s.strings.wc_walletconnect_subtitle}</EdgeText>
+        <MainButton
+          label={s.strings.wc_walletconnect_new_connection_button}
+          type="secondary"
+          marginRem={[1, 0.5]}
+          onPress={() => handleNewConnectionPress()}
+          alignSelf="center"
+        />
+        <EdgeText style={styles.listTitle}>{s.strings.wc_walletconnect_active_connections}</EdgeText>
+        <View style={styles.list}>
+          {dAppConnections.map((dAppConnection: WcConnectionInfo) => (
```

Comment:
dAppConnections.map((dAppConnection: WcConnectionInfo, index)

use 'index' for 122

Reply:
PR1